libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -846,32 +846,37 @@ class MapboxNavigation(
 
     private fun createTilesConfig(): TilesConfig {
         // TODO StrictMode may report a violation as we're creating a File from the Main
-        val offlineFilesPath = OnboardRouterFiles(navigationOptions.applicationContext, logger)
-            .absolutePath(navigationOptions.onboardRouterOptions)
-        val dataset = StringBuilder().apply {
-            append(navigationOptions.onboardRouterOptions.tilesDataset)
-            append("/")
-            append(navigationOptions.onboardRouterOptions.tilesProfile)
-        }.toString()
-
-        return TilesConfig(
-            offlineFilesPath,
-            null,
-            null,
-            null,
-            THREADS_COUNT,
-            TileEndpointConfiguration(
-                navigationOptions.onboardRouterOptions.tilesUri.toString(),
-                dataset,
-                navigationOptions.onboardRouterOptions.tilesVersion,
-                navigationOptions.accessToken ?: "",
-                USER_AGENT,
-                BuildConfig.NAV_NATIVE_SDK_VERSION,
-                NativeSkuTokenProvider(
-                    MapboxNavigationAccounts.getInstance(navigationOptions.applicationContext)
+        navigationOptions.run {
+            val offlineFilesPath = OnboardRouterFiles(applicationContext, logger)
+                .absolutePath(onboardRouterOptions)
+            val dataset = StringBuilder().apply {
+                append(onboardRouterOptions.tilesDataset)
+                append("/")
+                append(onboardRouterOptions.tilesProfile)
+            }.toString()
+            val host = StringBuilder().apply {
```

Comment:
to avoid string "://"
```suggestion
            val host = onboardRouterOptions.tilesUri.let { uri -> URI(uri.scheme, uri.host, null, null, null) }.toString()
```

Reply:
fixed